lib/cronofy/client.rb

```diff
@@ -525,6 +525,39 @@ def revoke_authorization
       @auth.revoke!
     end
 
+    # Public: Requests elevated permissions for a set of calendars.
+    #
+    # permissions  - An Array of calendar permission hashes to set on the each
+    #                hash must contain symbols for both `calendar_id` and
+    #                `permission_level`
+    # redirect_uri - A uri to redirect the end user back to after they have
+    #                either granted or rejected the request for elevated
+    #                permission.
+    #
+    # In the case of normal accounts:
+    # After making this call the end user will have to grant the extended
+    # permissions to their calendar via rhe url returned from the response.
+    #
+    # In the case of service accounts:
+    # After making this call the exteneded permissions will be granted provided
+    # the relevant scope has been granted to the account
+    #
+    # Returns a extended permissions response.
+    #
+    # Raises Cronofy::AuthenticationFailureError if the client ID and secret are
+    # not valid.
+    def elevated_permission(permissions, redirect_uri = nil)
```

Comment:
I think this will feel a bit awkward if you call it inline:

```ruby
client.elevated_permission([{ calendar_id: "abc", permission_level: :unrestricted }, { calendar_id: "def", permission_level: :unrestricted }], "https://redirect.example.com")

# or

client.elevated_permission(
  [
    { calendar_id: "abc", permission_level: :unrestricted },
    { calendar_id: "def", permission_level: :unrestricted }
  ],
  "https://redirect.example.com"
)
```

An args `Hash` perhaps helps this?

```ruby
client.elevated_permission(
  permissions: [
    { calendar_id: "abc", permission_level: :unrestricted },
    { calendar_id: "def", permission_level: :unrestricted }
  ],
  redirect_uri: "https://redirect.example.com"
)
```

WDYT?